backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/process/impl/ProcessRestController.java

```diff
@@ -0,0 +1,232 @@
+			@Nullable @RequestBody final RunProcessRequest request)
+	{
+
+		final Optional<AdProcessId> processId = getProcessIdIfRunnable(processValue);
+
+		if (!processId.isPresent())
+		{
+			return ResponseEntity.badRequest().body(Message.of("The requested process is not runnable or it doesn't exist!"));
+		}
+
+		final boolean cannotRunProcess = !permissionServiceFactory.createPermissionService().canRunProcess(processId.get());
+
+		if (cannotRunProcess)
+		{
+			return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+		}
+
+		final ProcessInfo.ProcessInfoBuilder processInfoBuilder = ProcessInfo.builder();
+		processInfoBuilder.setAD_Process_ID(processId.get());
+
+		final boolean parametersPresent = request != null && !Check.isEmpty(request.getProcessParameters());
+
+		if (parametersPresent)
+		{
+			request.getProcessParameters().forEach(param -> processInfoBuilder.addParameter(param.getName(), param.getValue()));
+		}
+
+		final ProcessExecutionResult processExecutionResult = processInfoBuilder
+				.buildAndPrepareExecution()
+				.executeSync()
+				.getResult();
+
+		return getResponse(processExecutionResult);
+	}
+
+	@GetMapping("/available")
+	public ResponseEntity<GetAvailableProcessesResponse> getAllProcesses()
+	{
+		final List<ProcessBasicInfo> processList = processService.getProcessesByType(ProcessType.getTypesRunnableFromAppRestController());
+
+		final List<JSONProcessBasicInfo> jsonProcessList = processList
+				.stream()
+				.map(this::buildJSONProcessBasicInfo)
+				.collect(Collectors.toList());
+
+		final GetAvailableProcessesResponse response = GetAvailableProcessesResponse
+				.builder()
+				.availableProcesses(jsonProcessList)
+				.build();
+
+		return ResponseEntity
+				.ok()
+				.contentType(MediaType.APPLICATION_JSON_UTF8)
+				.body(response);
+	}
+
+	private ResponseEntity<?> getResponse(@NonNull final ProcessExecutionResult processExecutionResult)
+	{
+		if (processExecutionResult.isError())
+		{
+			return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(Message.of("Ups! Looks like something went wrong on our end..."));
+		}
+		else if (Check.isNotBlank(processExecutionResult.getJsonResult()))
+		{
+			return ResponseEntity
+					.ok()
+					.contentType(MediaType.APPLICATION_JSON_UTF8)
+					.body(processExecutionResult.getJsonResult());
+		}
+		else if (Check.isNotBlank(processExecutionResult.getReportFilename()))
+		{
+			final String contentType = Check.isNotBlank(processExecutionResult.getReportContentType())
+					? processExecutionResult.getReportContentType()
+					: MediaType.APPLICATION_OCTET_STREAM_VALUE;
+
+			return ResponseEntity.ok()
+					.contentType(MediaType.parseMediaType(contentType))
+					.header(HttpHeaders.CONTENT_DISPOSITION,"attachment; filename=\"" + processExecutionResult.getReportFilename() + "\"")
+					.body(processExecutionResult.getReportData());
+		}
+		else
+		{
+			//this should never happen
+			return ResponseEntity.noContent().build();
```

Comment:
if it should never happen, i suggest you throw an exception or do `Check.fail(...)`